midenc-compile/src/stages/link.rs

```diff
@@ -21,14 +23,19 @@ impl Stage for LinkerStage {
         _analyses: &mut AnalysisManager,
         session: &Session,
     ) -> CompilerResult<Self::Output> {
-        if session.should_link() {
-            let mut builder = hir::ProgramBuilder::new(&session.diagnostics);
-            for module in input.into_iter() {
-                builder.add_module(module)?;
+        match session.project_type {
```

Comment:
My initial reason was that `session.should_link()` calls `OutputTypes::should_link()` which is https://github.com/0xPolygonMiden/compiler/blob/820cd76ef371460915d0890b066640799b6781eb/midenc-session/src/outputs.rs#L251
so effectively, the backend builds only `Program` for all `OutputType::Masm` output types because an empty entrypoint case in the program linker was a `todo!()` with a link to some VM issue, so there was no way for me to build a program/library without an entrypoint.

I suppose you are describing the approach you introduced in #181. I should revisit my changes and test the original issue again.

Reply:
I see, I think we can keep this for now then, just with a note that it's a temporary workaround for that specific issue. I'll revisit the program linker sometime in the next few days/weeks and either fix that specific issue, or figure out a different longer term solution.

Reply:
I tested it on top of the new backend changes and the issue is still there. I agree, let's keep it for now. I added the comment explaining the workaround.